prog.go

```diff
@@ -409,59 +415,45 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions) (*Program, er
 		}
 	}
 
-	// The caller requested a specific verifier log level. Set up the log buffer
-	// so that there is a chance of loading the program in a single shot.
-	logSize := internal.Between(opts.LogSizeStart, minVerifierLogSize, maxVerifierLogSize)
 	var logBuf []byte
-	if !opts.LogDisabled && opts.LogLevel != 0 {
-		logBuf = make([]byte, logSize)
-		attr.LogLevel = opts.LogLevel
-		attr.LogSize = uint32(len(logBuf))
-		attr.LogBuf = sys.NewSlicePointer(logBuf)
-	}
+	var fd *sys.FD
 
-	for {
-		var fd *sys.FD
+	if opts.LogDisabled {
+		// Loading with logging disabled should never retry.
 		fd, err = sys.ProgLoad(attr)
 		if err == nil {
-			return &Program{unix.ByteSliceToString(logBuf), fd, spec.Name, "", spec.Type}, nil
-		}
-
-		if opts.LogDisabled {
-			break
+			return &Program{"", fd, spec.Name, "", spec.Type}, nil
 		}
```

Comment:
You can return here, all of the other code only deals with the log being present. Then the else can go and we can unindent by one leve.

Reply:
What do you mean? The code after the `else {}` deals with wrapping EPERM, EFAULT, etc.